src/main/java/zeenea/connector/property/PropertyDefinition.java

```diff
@@ -38,18 +35,32 @@ public abstract class PropertyDefinition {
    * @param label the label of the property definition
    */
   PropertyDefinition(@NotNull String code, @NotNull PropertyType type, @Nullable String label) {
-    this.code = Objects.requireNonNull(code);
+    this(label == null ? Objects.requireNonNull(code) : label, type);
+  }
+
+  /**
+   * Constructs a PropertyDefinition instance with the specified name and type.
+   *
+   * @param name the name of the property definition
+   * @param type the type of the property definition
+   */
+  PropertyDefinition(@NotNull String name, @NotNull PropertyType type) {
+    this.name = Objects.requireNonNull(name);
     this.type = Objects.requireNonNull(type);
-    this.label = label;
   }
 
   /**
    * Gets the code of the property definition.
    *
    * @return the code of the property definition
    */
+  @Deprecated(since = "2.3.0", forRemoval = true)
```

Comment:
Je te propose un message de dépréciation plus "actionable" et clair quant au futur du SDK :
```suggestion
  
  @Deprecated(
      since =
          "Deprecated since version 2.3.0, use getName() instead. Scheduled for removal in version 3.0.0.",
      forRemoval = true)
```